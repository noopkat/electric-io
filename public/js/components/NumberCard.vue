<template>
  <p class="number" v-bind:style="numberStyle">{{ number }}</p>
</template>

<script>
import { evaluatePath } from "../lib/messagePropertyEvaluation.js";

export default {
  name: "number-card",
  props: {
    tile: {
      default: () => ({})
    },
    blockSize: {
      default: () => []
    },
    messages: {
      default: () => []
    }
  },
  data: function() {
    return {
      number: 0
    };
  },
  watch: {
    messages: function() {
      const lastMessage = this.messages.pop();
      if (lastMessage) {
        const value = evaluatePath(this.tile.property, lastMessage);
        this.number = value ? parseFloat(value).toFixed(1) : "";
      } else {
        this.number = 0;
      }
    }
  },
  computed: {
    numberStyle: function() {
      return { color: this.tile.textColor };
    }
  }
};
</script>
