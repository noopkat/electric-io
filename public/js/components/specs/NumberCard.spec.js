import { shallowMount } from "@vue/test-utils";
import NumberCard from "../NumberCard";
import axe from "axe-core";

describe("NumberCard", () => {
  test("component can mount", () => {
    const wrapper = shallowMountNumberCard();

    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("renders with color and number value", () => {
    const wrapper = shallowMountNumberCard();

    expect(wrapper.html()).toMatchSnapshot();
  });

  test("the messages watch method", () => {
    const spy = jest.spyOn(NumberCard.watch, "messages");
    const wrapper = shallowMount(NumberCard, {
      propsData: {
        messages: []
      }
    });

    expect(spy).toHaveBeenCalledTimes(0);

    wrapper.setProps({
      messages: [
        {
          deviceId: "BU2802",
          enqueuedTime: "2019-06-03T11:45:10.125Z",
          humidity: 32.800208338,
          temperature: 45.13494407
        },
        {
          deviceId: "AZ3166",
          enqueuedTime: "2019-06-03T11:33:10.125Z",
          humidity: 7.053375767532866,
          temperature: 31.599309710235097
        }
      ]
    });

    expect(spy).toHaveBeenCalled();
  });

  test("verify component is accessible", () => {
    const wrapper = shallowMountNumberCard();

    axe.run(wrapper, (err, { violations }) => {
      expect(err).toBe(null);
      expect(violations).toHaveLength(0);
      done();
    });
  });
});

const mountingConfiguration = {
  propsData: {
    tile: {
      textColor: "blue"
    }
  },
  data: () => ({
    number: 1
  })
};

function shallowMountNumberCard() {
  return shallowMount(NumberCard, mountingConfiguration);
}
