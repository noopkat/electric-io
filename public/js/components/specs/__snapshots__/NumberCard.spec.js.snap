// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NumberCard renders with color and number value 1`] = `"<p class=\\"number\\" style=\\"color: blue;\\">1</p>"`;
